db/hints/sync_point.cc

```diff
@@ -67,16 +69,30 @@ static std::vector<sync_point::shard_rps> decode_one_type_v1(uint16_t shard_coun
     return ret;
 }
 
+static uint64_t get_checksum(signed char* data, size_t size) {
+    xx_hasher h;
+    h.update(reinterpret_cast<const char*>(data), size);
+    return h.finalize_uint64();
+}
+
 sync_point sync_point::decode(sstring_view s) {
     bytes raw = base64_decode(s);
     if (raw.empty()) {
         throw std::runtime_error("Could not decode the sync point - not a valid hex string");
     }
-    if (raw[0] != 1) {
+
+    if (raw[0] == 2) {
+        seastar::simple_memory_input_stream in_checksum{reinterpret_cast<const char*>(raw.end()) - sizeof(uint64_t), sizeof(uint64_t)};
```

Comment:
Please introduce a constant somewhere somewhere high in the file (e.g. right after the comment with the  description of the serialization format):

```c++
static constexpr size_t checksum_size = sizeof(uint64_t);
```

and then use it in instead of `sizeof(uint64_t)`. It will be more readable that way.